test/rules/max-nesting-depth/specifiedThreshold/test.ts.lint

```diff
@@ -0,0 +1,270 @@
+        }
+    }
+}
+
+function validForIn() {
+    for(let i in [0,1,2]) {
+        for(let i in [0,1,2]) { }
+    }
+}
+
+function invalidForIn() {
+    for(let i in [0,1,2]) {
+        for(let i in [0,1,2]) {
+            for(let i in [0,1,2]) { }
+            ~~~~~~~~~~~~~~~~~~~~~~~~~ [The function invalidForIn has a nesting depth of at least 3 which is higher than the threshold of 2]
+        }
+    }
+}
+
+function validForOf() {
+    for(let i of [0,1,2]) {
+        for(let i of [0,1,2]) { }
+    }
+}
+
+function invalidForOf() {
+    for(let i of [0,1,2]) {
+        for(let i of [0,1,2]) {
+            for(let i of [0,1,2]) { }
+            ~~~~~~~~~~~~~~~~~~~~~~~~~ [The function invalidForOf has a nesting depth of at least 3 which is higher than the threshold of 2]
+        }
+    }
+}
+
+function validIf() {
+    if(true) {
+        if(true) { } else { }
+    } else {
+        if(true) { } else { }
+    }
+}
+
+function invalidIf() {
+    if(true) {
+        if(true) {
+            if(true) { } else { }
+            ~~~~~~~~~~~~~~~~~~~~~ [The function invalidIf has a nesting depth of at least 3 which is higher than the threshold of 2]
+        } else {
+            if(true) { } else { }
+            ~~~~~~~~~~~~~~~~~~~~~ [The function invalidIf has a nesting depth of at least 3 which is higher than the threshold of 2]
+        }
+    }
+}
+
+// Test that the rule behaves correctly for nested functions.
+
+// Valid outer function, invalid inner function.
+function outer1() {
+    const condition = false ? "true" : false ? "true" : "false";
+    function inner1() {
+        const condition = false ? "true" : false ? "true" : false ? "true" : "false";
+                                                            ~~~~~~~~~~~~~~~~~~~~~~~~  [The function inner1 has a nesting depth of at least 3 which is higher than the threshold of 2]
+    }
+}
+
+// Invalid outer function, valid inner function.
+function outer2() {
+    const condition = false ? "true" : false ? "true" : false ? "true" : "false";
+                                                        ~~~~~~~~~~~~~~~~~~~~~~~~  [The function outer2 has a nesting depth of at least 3 which is higher than the threshold of 2]
+    function inner2() {
+        const condition = false ? "true" : false ? "true" : "false";
+    }
+}
+
+// Both valid.
+function outer3() {
+    const condition = false ? "true" : false ? "true" : "false";
+    function inner3() {
+        const condition = false ? "true" : false ? "true" : "false";
+    }
+}
+
+// Both invalid.
+function outer4() {
+    const condition = false ? "true" : false ? "true" : false ? "true" : "false";
+                                                        ~~~~~~~~~~~~~~~~~~~~~~~~  [The function outer4 has a nesting depth of at least 3 which is higher than the threshold of 2]
+   function inner4() {
+        const condition = false ? "true" : false ? "true" : false ? "true" : "false";
```

Comment:
surprised that the error here is at the same depth as line 264, given that this function is nested one level deeper. why is that?

Reply:
It is so because, as ESLint does, I reset the depth counter each time a new function is entered.

Reply:
Does it make sense ?